core/src/main/java/org/apache/iceberg/jdbc/JdbcClientPool.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.jdbc;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.SQLNonTransientConnectionException;
+import java.sql.SQLTimeoutException;
+import java.sql.SQLTransientConnectionException;
+import java.sql.SQLWarning;
+import java.util.Properties;
+import org.apache.iceberg.ClientPool;
+import org.apache.iceberg.exceptions.UncheckedSQLException;
+
+public class JdbcClientPool extends ClientPool<Connection, SQLException> {
+
+  private final String dbUrl;
+  private final Properties dbProperties;
+
+  JdbcClientPool(String dbUrl, Properties props) {
```

Comment:
I don't think that Iceberg uses `Properties` anywhere, so this should probably accept `Map<String, String>` and pass that to JDBC as properties if necessary.

Also, config properties don't need config context like `iceberg.jdbc` because that context is already dependent on how the catalog is configured. For example, Spark catalogs will use `spark.sql.catalog.catalog_name.uri` for the connection URI already. So the property keys here should just be the standard ones defined in [`CatalogProperties`](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/CatalogProperties.java).

Reply:
i see updated it, 

@rdblue what do you think about passing additional jdbc parameters with following prefix? is it okay or there shuld be no prefix?

`public static final String JDBC_PARAM_PREFIX = "connection.parameter.";`

current setup will be like 
```
spark.sql.catalog.catalog_name.uri=jdbc:postgresql://localhost:5432/mydatabase
spark.sql.catalog.catalog_name.connection.parameter.user=myuser
spark.sql.catalog.catalog_name.connection.parameter.password=mypass
spark.sql.catalog.catalog_name.connection.parameter.ssl=true
spark.sql.catalog.catalog_name.connection.parameter.param1=val1
spark.sql.catalog.catalog_name.connection.parameter.param2=val2
```